app/javascript/packs/base.js

```diff
@@ -14,14 +14,18 @@ function getPageEntries() {
   });
 }
 
-const menus = [...document.getElementsByClassName('js-hamburger-trigger')];
-const moreMenus = [...document.getElementsByClassName('js-nav-more-trigger')];
+let menus = [...document.getElementsByClassName('js-hamburger-trigger')];
+let moreMenus = [...document.getElementsByClassName('js-nav-more-trigger')];
 
 getInstantClick().then((spa) => {
   spa.on('change', () => {
+    let menus = [...document.getElementsByClassName('js-hamburger-trigger')];
+    let moreMenus = [...document.getElementsByClassName('js-nav-more-trigger')];
```

Comment:
Refactor (blocking): The variables are already defined in the outer scope.
```suggestion
    menus = [...document.getElementsByClassName('js-hamburger-trigger')];
    moreMenus = [...document.getElementsByClassName('js-nav-more-trigger')];
```

Refactor (suggestion): What would be better is to make all this logic a function that gets called by `InstantClick` and on page load. This is a common pattern that occurs in the code base. We want code to run on page load and also when InstantClick changes the page.

e.g.

```js
function initializeMenus() {
  const { currentPage } = document.getElementById('page-content').dataset;
  const menus = [...document.getElementsByClassName('js-hamburger-trigger')];
  const moreMenus = [...document.getElementsByClassName('js-nav-more-trigger')];
  const memberMenu = document.getElementById('crayons-header__menu');
  const menuNavButton = document.getElementById('member-menu-button');

  setCurrentPageIconLink(currentPage, getPageEntries());
  initializeMobileMenu(menus, moreMenus);
  initializeTouchDevice(memberMenu, menuNavButton);
}

getInstantClick().then((spa) => {
  spa.on('change', initializeMenus);
});

initializeMenus();
```

Not sure on the naming of that though as it's the menus being set as well as the current page icon.

Reply:
Ahh that makes much more sense. I actually looked at another code example but didn't think to refactor it to the pattern you showed. Will update!